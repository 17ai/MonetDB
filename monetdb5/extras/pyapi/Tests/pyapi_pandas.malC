


# read data from a CSV file using the Python PANDAS library and store it in a BAT
# requires python-pandas to be installed
# CSV location is hardcoded (sorry) ----->---------->----------->-------->------vvvv
r:bat[:oid,:int] := pyapi.eval(nil:ptr, "import pandas as pd\ndf = pd.read_csv(\"/home/mytherin/Programs/MonetDB/monetdb5/extras/pyapi/Tests/bikedata.csv\")\nreturn([list(df['Berri 1'])])");
io.print(r);
