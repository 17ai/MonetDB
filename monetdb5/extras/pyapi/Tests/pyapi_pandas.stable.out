stdout of test 'pyapi_pandas` in directory 'monetdb5/extras/pyapi` itself:


# 19:41:50 >  
# 19:41:50 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=31934" "--set" "mapi_usock=/var/tmp/mtest-7261/.s.monetdb.31934" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/usr/local/var/MonetDB/mTests_monetdb5_extras_pyapi" "--set" "mal_listing=2" "--set" "embedded_r=true" "--set" "embedded_py=true"
# 19:41:50 >  

# MonetDB 5 server v11.20.0
# This is an unreleased version
# Serving database 'mTests_monetdb5_extras_pyapi', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 7.684 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2015 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://mytherin-N750JV:31934/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-7261/.s.monetdb.31934
# MonetDB/SQL module loaded
# MonetDB/Python module loaded
# MonetDB/R   module loaded

Ready.

# 19:41:50 >  
# 19:41:50 >  "mclient" "-lmal" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-7261" "--port=31934"
# 19:41:50 >  

#io.print(r);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	35	]
[ 1@0,	83	]
[ 2@0,	135	]
[ 3@0,	144	]
[ 4@0,	197	]
[ 5@0,	146	]
[ 6@0,	98	]
[ 7@0,	95	]
[ 8@0,	244	]
[ 9@0,	397	]
[ 10@0,	273	]
[ 11@0,	157	]
[ 12@0,	75	]
[ 13@0,	32	]
[ 14@0,	54	]
[ 15@0,	168	]
[ 16@0,	155	]
[ 17@0,	139	]
[ 18@0,	191	]
[ 19@0,	161	]
[ 20@0,	53	]
[ 21@0,	71	]
[ 22@0,	210	]
[ 23@0,	299	]
[ 24@0,	334	]
[ 25@0,	306	]
[ 26@0,	91	]
[ 27@0,	80	]
[ 28@0,	87	]
[ 29@0,	219	]
[ 30@0,	186	]
[ 31@0,	138	]
[ 32@0,	217	]
[ 33@0,	174	]
[ 34@0,	84	]
[ 35@0,	72	]
[ 36@0,	248	]
[ 37@0,	316	]
[ 38@0,	271	]
[ 39@0,	342	]
[ 40@0,	303	]
[ 41@0,	71	]
[ 42@0,	78	]
[ 43@0,	211	]
[ 44@0,	318	]
[ 45@0,	307	]
[ 46@0,	386	]
[ 47@0,	332	]
[ 48@0,	220	]
[ 49@0,	169	]
[ 50@0,	303	]
[ 51@0,	441	]
[ 52@0,	375	]
[ 53@0,	397	]
[ 54@0,	243	]
[ 55@0,	62	]
[ 56@0,	78	]
[ 57@0,	119	]
[ 58@0,	195	]
[ 59@0,	242	]
[ 60@0,	92	]
[ 61@0,	143	]
[ 62@0,	82	]
[ 63@0,	107	]
[ 64@0,	155	]
[ 65@0,	269	]
[ 66@0,	438	]
[ 67@0,	348	]
[ 68@0,	371	]
[ 69@0,	182	]
[ 70@0,	380	]
[ 71@0,	802	]
[ 72@0,	442	]
[ 73@0,	469	]
[ 74@0,	724	]
[ 75@0,	423	]
[ 76@0,	681	]
[ 77@0,	1940	]
[ 78@0,	1821	]
[ 79@0,	2481	]
[ 80@0,	2829	]
[ 81@0,	2195	]
[ 82@0,	2115	]
[ 83@0,	753	]
[ 84@0,	520	]
[ 85@0,	968	]
[ 86@0,	1049	]
[ 87@0,	435	]
[ 88@0,	878	]
[ 89@0,	1157	]
[ 90@0,	980	]
[ 91@0,	662	]
[ 92@0,	1937	]
[ 93@0,	2416	]
[ 94@0,	2211	]
[ 95@0,	2424	]
[ 96@0,	1633	]
[ 97@0,	1208	]
[ 98@0,	1164	]
[ 99@0,	828	]
[ 100@0,	2183	]
[ 101@0,	2328	]
[ 102@0,	3064	]
[ 103@0,	3341	]
[ 104@0,	2890	]
[ 105@0,	2554	]
[ 106@0,	3643	]
[ 107@0,	3539	]
[ 108@0,	3570	]
[ 109@0,	4231	]
[ 110@0,	2087	]
[ 111@0,	533	]
[ 112@0,	1853	]
[ 113@0,	623	]
[ 114@0,	1810	]
[ 115@0,	2966	]
[ 116@0,	2751	]
[ 117@0,	1986	]
[ 118@0,	1684	]
[ 119@0,	1970	]
[ 120@0,	3610	]
[ 121@0,	1986	]
[ 122@0,	3724	]
[ 123@0,	3698	]
[ 124@0,	2511	]
[ 125@0,	3492	]
[ 126@0,	3411	]
[ 127@0,	5552	]
[ 128@0,	1241	]
[ 129@0,	3297	]
[ 130@0,	2755	]
[ 131@0,	4639	]
[ 132@0,	3854	]
[ 133@0,	2741	]
[ 134@0,	6189	]
[ 135@0,	3964	]
[ 136@0,	4947	]
[ 137@0,	5351	]
[ 138@0,	5980	]
[ 139@0,	4732	]
[ 140@0,	5255	]
[ 141@0,	5129	]
[ 142@0,	2315	]
[ 143@0,	5974	]
[ 144@0,	6485	]
[ 145@0,	5697	]
[ 146@0,	4974	]
[ 147@0,	4396	]
[ 148@0,	4268	]
[ 149@0,	3154	]
[ 150@0,	6459	]
[ 151@0,	5104	]
[ 152@0,	6097	]
[ 153@0,	943	]
[ 154@0,	2755	]
[ 155@0,	2717	]
[ 156@0,	5842	]
[ 157@0,	6037	]
[ 158@0,	6246	]
[ 159@0,	4169	]
[ 160@0,	5164	]
[ 161@0,	5112	]
[ 162@0,	6206	]
[ 163@0,	3361	]
[ 164@0,	6180	]
[ 165@0,	6908	]
[ 166@0,	7077	]
[ 167@0,	5421	]
[ 168@0,	4638	]
[ 169@0,	5921	]
[ 170@0,	5382	]
[ 171@0,	5713	]
[ 172@0,	5183	]
[ 173@0,	5398	]
[ 174@0,	3753	]
[ 175@0,	3341	]
[ 176@0,	2245	]
[ 177@0,	3327	]
[ 178@0,	3141	]
[ 179@0,	6064	]
[ 180@0,	5770	]
[ 181@0,	4738	]
[ 182@0,	4758	]
[ 183@0,	4144	]
[ 184@0,	6712	]
[ 185@0,	5153	]
[ 186@0,	6672	]
[ 187@0,	5958	]
[ 188@0,	5420	]
[ 189@0,	4756	]
[ 190@0,	5661	]
[ 191@0,	6500	]
[ 192@0,	6424	]
[ 193@0,	6179	]
[ 194@0,	5518	]
[ 195@0,	4206	]
[ 196@0,	3035	]
[ 197@0,	4827	]
[ 198@0,	2756	]
[ 199@0,	6180	]
[ 200@0,	6309	]
[ 201@0,	5813	]
[ 202@0,	5092	]
[ 203@0,	4971	]
[ 204@0,	3877	]
[ 205@0,	5243	]
[ 206@0,	6104	]
[ 207@0,	5560	]
[ 208@0,	5759	]
[ 209@0,	4105	]
[ 210@0,	4230	]
[ 211@0,	5225	]
[ 212@0,	5415	]
[ 213@0,	4638	]
[ 214@0,	5715	]
[ 215@0,	5577	]
[ 216@0,	4223	]
[ 217@0,	1864	]
[ 218@0,	5240	]
[ 219@0,	5997	]
[ 220@0,	5498	]
[ 221@0,	4127	]
[ 222@0,	2414	]
[ 223@0,	2453	]
[ 224@0,	3995	]
[ 225@0,	4931	]
[ 226@0,	5333	]
[ 227@0,	4297	]
[ 228@0,	6062	]
[ 229@0,	4236	]
[ 230@0,	4427	]
[ 231@0,	4644	]
[ 232@0,	5335	]
[ 233@0,	5792	]
[ 234@0,	6529	]
[ 235@0,	5437	]
[ 236@0,	5690	]
[ 237@0,	4242	]
[ 238@0,	3964	]
[ 239@0,	4314	]
[ 240@0,	5817	]
[ 241@0,	6327	]
[ 242@0,	6003	]
[ 243@0,	4106	]
[ 244@0,	3934	]
[ 245@0,	3698	]
[ 246@0,	4179	]
[ 247@0,	4225	]
[ 248@0,	5655	]
[ 249@0,	5883	]
[ 250@0,	6186	]
[ 251@0,	2155	]
[ 252@0,	3312	]
[ 253@0,	5077	]
[ 254@0,	6015	]
[ 255@0,	6349	]
[ 256@0,	6520	]
[ 257@0,	5216	]
[ 258@0,	3341	]
[ 259@0,	3635	]
[ 260@0,	5299	]
[ 261@0,	2530	]
[ 262@0,	4653	]
[ 263@0,	5260	]
[ 264@0,	4022	]
[ 265@0,	1521	]
[ 266@0,	2314	]
[ 267@0,	4553	]
[ 268@0,	5038	]
[ 269@0,	3948	]
[ 270@0,	5119	]
[ 271@0,	4652	]
[ 272@0,	1896	]
[ 273@0,	876	]
[ 274@0,	3255	]
[ 275@0,	5139	]
[ 276@0,	4685	]
[ 277@0,	4034	]
[ 278@0,	4151	]
[ 279@0,	1304	]
[ 280@0,	1580	]
[ 281@0,	1854	]
[ 282@0,	4787	]
[ 283@0,	3115	]
[ 284@0,	3746	]
[ 285@0,	3169	]
[ 286@0,	1783	]
[ 287@0,	587	]
[ 288@0,	3292	]
[ 289@0,	3739	]
[ 290@0,	4098	]
[ 291@0,	4671	]
[ 292@0,	1313	]
[ 293@0,	2011	]
[ 294@0,	1277	]
[ 295@0,	3650	]
[ 296@0,	4177	]
[ 297@0,	3744	]
[ 298@0,	3735	]
[ 299@0,	4290	]
[ 300@0,	1857	]
[ 301@0,	1310	]
[ 302@0,	2919	]
[ 303@0,	2887	]
[ 304@0,	2634	]
[ 305@0,	2405	]
[ 306@0,	1582	]
[ 307@0,	844	]
[ 308@0,	966	]
[ 309@0,	2247	]

# 19:41:50 >  
# 19:41:50 >  "Done."
# 19:41:50 >  

